### ISSUE TITLE
Staging Review finding: Prescription refill not confirmed or reversible

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#84224]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** MHV-Medications
**Product:** MHV Medications
**Feature:** Medications

## Findings details
**VA.gov Experience Standard - issue:** [User can't review, edit, or confirm their information before form submission.](https://design.va.gov/about/experience-standards/#user-can-review-edit-or-confirm-their-information-before-form-submission)
**VA.gov Experience Standard - category:** [Credibility](https://design.va.gov/about/experience-standards/#category-credibility)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

This is a maybe-launch-blocker, but I need a little more information from the team to help decide whether it's actually a launch-blocker. I'm a little concerned about the prescription refill action potentially violating [WCAG 3.3.4](https://www.w3.org/WAI/WCAG22/Understanding/error-prevention-legal-financial-data.html):

> For Web pages that cause legal commitments or financial transactions for the user to occur, that modify or delete user-controllable data in data storage systems, or that submit user test responses, at least one of the following is true:
> 
> **Reversible**
> Submissions are reversible.
> 
> **Checked**
> Data entered by the user is checked for input errors and the user is provided an opportunity to correct them.
> 
> **Confirmed**
> A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission.

In the current flow, none of those three criteria are met. Once you select the medications you want to refill, it's a single-click action to initiate the refill with no opportunity to confirm prior to submission or to reverse/correct after submission.

Since medications may [require a copayment](https://www.va.gov/healthbenefits/resources/publications/hbco/hbco_copayments.asp), that could be considered a financial transaction --- although I'm not sure if that's actually an issue here. Is any financial transaction initiated by requesting a refill? Or does the financial transaction only happen at the pharmacy at the time of pickup?

**Update:** Received this context during the staging review meeting:
> About the potential launch blocker for money transactions: Refills requests don't always get processed that day and may never get processed so there is no direct money transactions that happen when they click refill but the user would get billed once they got the medication. I can keep digging into this if need be or if that helps.

Based on that, I don't think WCAG 3.3.4 applies here. Downgrading the ticket and removing the `launch-blocking` label.

I will note that the page may not provide full context for what happens when refilling prescriptions, so a reminder to Veterans that a copay may apply might be worth exploring. 

## Recommended action

If the financial transaction is reversible at the pharmacy counter (refuse to pay, no consequences besides not getting their medications) then this may be fine under WCAG.

If the financial transaction isn't reversible, then some mechanism needs to be provided to either:
- Confirm the action (eg. "Are you sure?" modal after clicking the refill button)
- Reverse the action (eg. "Didn't mean to request that refill? [button]Cancel your refill request[/button]")

But regardless of whether this is a WCAG violation or requires any changes to functionality, I think a content change would be beneficial here to remind users that a copay may apply.

**Update:** See note above. I'd recommend exploring a content change to see if the full flow can be made more clear, but no immediate accessibility concerns.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** [WCAG 3.3.4 Error Prevention (Legal, Financial, Data) - Level AA](https://www.w3.org/TR/WCAG21/#error-prevention-legal-financial-data)
- **Modality:** N/A
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by briandeconinck** [comment]
Updated ticket and removed labels based on context provided during the staging review meeting. Thank you!

**Comment by IdeaTrail** [comment]
UCD will design new content.

**Comment by shiragoodman** [comment]
Hi @IdeaTrail  - I am reopening this issue as we kindly request that Staging Review tickets not be closed until the issue has been resolved. Please let me know if you have any questions - ty!
